File: python/2차 1급 4_initial_code.py
#다음과 같이 import를 사용할 수 있습니다.
#import math

def solution(arr, K):
    arr_len = len(arr)
    answer = 0

    for a in range(0, arr_len - 2):
        for b in range(a + 1, arr_len - 1):
            for c in range(b + 1, arr_len):
                if (arr[a] + arr[b] + arr[c]) % K != 0:
                    continue

                answer += 1

    return answer

#아래는 테스트케이스 출력을 해보기 위한 코드입니다.
arr = [1, 2, 3, 4, 5]
K = 3
ret = solution(arr, K)

#[실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
print("solution 함수의 반환 값은 ", ret, " 입니다.")

File: c/2차 1급 4_initial_code.c
// 다음과 같이 include를 사용할 수 있습니다.
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int arr[], int arr_len, int K) {
    int answer = 0;

    for (int a = 0; a < arr_len - 2; a++) {
        for (int b = a + 1; b < arr_len - 1; b++) {
            for (int c = b + 1; c < arr_len; c++) {
                if ((arr[a] + arr[b] + arr[c]) % K != 0) {
                    continue;
                }

                answer++;
            }
        }
    }

    return answer;
}

// 아래는 테스트케이스 출력을 해보기 위한 main 함수입니다.
int main() {
    int arr[5] = {1, 2, 3, 4, 5};
    int arr_len = 5;
    int K = 3;
    int ret = solution(arr, arr_len, K);

    // [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
    printf("solution 함수의 반환 값은 %d 입니다.\n", ret);
}